### ISSUE TITLE
iframe page for embedding in 3rd party websites

### ISSUE THREAD

**Issue by mattsayre** [Hack4Eugene/SpeedUpAmerica#283]
**Is your feature request related to a problem? Please describe.**
Looking to enable 3rd party branded websites to embed a speedup page, in their website. This will be a survey style form, very similar to how it is done on speedupamerica.com proper, meaning a sequence for taking a speed test of geolocation (use my browsers location/use my addres), then Residential/Mobile/Public, connection tyep (wired/wireless/etc), price, advertised speed. Then return speed results and comparison.

**Describe the solution you'd like**
iframe so that it can be branded by third parties. Only branding inside the frame will be "Powered by SpeedUpAmerica"

**Describe alternatives you've considered**
Javascript embed

**Additional context**
Don't need mapping, this feature is simply for collecting crowdsourced speed test results.

**Comment by webaissance** [comment]
I am working on this. Didn't see how to assign to myself.

**Comment by mattsayre** [comment]
@webaissance a few copy and other minor edits:
1) Remove the copy "Test Your Internet Speed Oregon"
2) Remove the copy "In 5 Steps"
3) Remove the question about connection type (wired/wifi/cellular) in its entirety.
4) Change Estimated price per month question to OPTIONAL.
5) Remove csv Export button from iframe, just need to know the direct link URL.

**Comment by webaissance** [comment]
Okay @mattsayre - I will work on those - give me till tomorrow mid-morning and I'll send these updates through.

**Comment by mattsayre** [comment]
Additional minor edits @webaissance :
1) Make question 3 (regarding price), optional. This will require the additional change of having question 4 come up to fill out in tandem with question 3.
2) The "i" hover over in question 3 floats up and is getting cut off (readability). Needs to float down, which should prevent the text from being cutoff.
3) Remove entire statement on results page, "This test helps...".

**Comment by mattsayre** [comment]
Additional minor edit @webaissance 

1. we need to change the language under the "I don't have internet" track, when they enter their address. The copy should be replaced with "I'd like to register my address as a location that doesn't have internet".

**Comment by webaissance** [comment]
Ok. I'll work on sending another PR tomorrow or tonight if possible

**Comment by webaissance** [comment]
Matt - I just sent a PR that takes care of all of the remaining tasks for
the iFrame issue.

Also - of note - in order to download data you have to inspect the result
page and unhide the button. I don't know of a way to trigger the
export just via a URL.

The export process can be developed over time more as needed.

Hope tomorrow goes great for the project!

-Dave

**Comment by webaissance** [comment]
@mattsayre I am working on these. I would say anything that's just cosmetic or type change is minor.  Anything that changes functionality is small but not minor - because it changes the logic and has the potential to break things and therefore requires more thorough testing. In any case - I will try to get these done asap and send through a PR.

**Comment by webaissance** [comment]
For 1) - the new Opt-in item - we should probably add this as a database field - for opt-in right? This would involve running the db migrator update again.
For now I will not do that - since this appears to be urgent - but can add it later if needed.

**Comment by webaissance** [comment]
Related to this - wouldn't each region have it's own disclaimer - so the link shouldn't be hardcoded in  - but a configuration for each region?

**Comment by mattsayre** [comment]
Thank you. Yes, let’s add that later.

**Comment by mattsayre** [comment]
Good thinking. Let’s also put that on the future roadmap and just hard code it for now.

**Comment by mattsayre** [comment]
@webaissance follow-up to our conversation about restoring IP address lookup capability for naming ISP's.

**Comment by mattsayre** [comment]
@webaissance add Spanish language version of iframe. Copy attached
[iframe language for translation FIO-es-US.docx](https://github.com/Hack4Eugene/SpeedUpAmerica/files/8633765/iframe.language.for.translation.FIO-es-US.docx)
Of note, Internationalization in Rails: https://guides.rubyonrails.org/i18n.html
Also of note, need to link out to the host website and their spanish language translation version of the Disclaimber.

.

**Comment by mattsayre** [comment]
Link to (test) Spanish translation of disclaimer https://test.fasterinternetoregon.org/es/disclaimer/

**Comment by mattsayre** [comment]
@webaissance - BUG - Scenario where user clicks “I do NOT have internet…”, then “Internet is not affordable”, then register my address… it currently initiates a speed test. There should NOT be a speed test initiated in this flow.

**Comment by webaissance** [comment]
Okay - what should it do instead? What should the button say?

**Comment by mattsayre** [comment]
When user selects "I do NOT have Internet access..." > Internet is not affordable, they should receive this confirmation message (and not initiate the speed test): 
 
"Thank you for your participation. Get more information on state and federal government discounted Internet access programs: https://www.oregon.gov/puc/pages/oregon-lifeline.aspx”
 
"Gracias por su participación. Obtenga más información sobre programas del gobierno estatal y federal que ofrecen acceso a Internet con descuento: https://www.oregon.gov/puc/pages/oregon-lifeline.aspx”

When user selects "I do NOT have Internet access..." >  Internet is not available..., they receive this confirmation message: 

"Thank you for your participation." 
"Gracias por su participación.”

**Comment by webaissance** [comment]
@mattsayre - I'm not clear on the process flow for different scenarios - particularly if no internet connection.
If no internet - do we still want to collect data - or just do nothing?
What about the "register this address as having no internet" option from before?

It would nice to have a flowchart of the scenarios.
Here are 2 flowcharts I made. I'm trying to understand if you want scenario A or B - or something else?

[SpeedUp-FlowChartA-2022-05-08.pdf](https://github.com/Hack4Eugene/SpeedUpAmerica/files/8648837/SpeedUp-FlowChartA-2022-05-08.pdf)

[SpeedUp-FlowChartB-2022-05-08.pdf](https://github.com/Hack4Eugene/SpeedUpAmerica/files/8648840/SpeedUp-FlowChartB-2022-05-08.pdf)

**Comment by mattsayre** [comment]
@webaissance yes, we do want to collect data from all form submissions (have and have not internet). Flow chart B is very close, made one edit, see attached
![flow B version 2](https://user-images.githubusercontent.com/49382923/167346156-a3556056-49b7-449a-8290-fa72be76e63b.png)

.

**Comment by webaissance** [comment]
Thanks @mattsayre  - now I'm getting clarity. I made an updated flowchart on SmartDraw.com. It allows us to collaborate on Flowcharts via shared links, 
So you can go in and make edits here as needed - 

[FlowChartC-2022-05-09.pdf](https://github.com/Hack4Eugene/SpeedUpAmerica/files/8650818/FlowChartC-2022-05-09.pdf)

I think if we collaborate on flow charts like this it will really help provide clarity and accuracy to the project and we can accelerate our progress.

I'm evaluating [SmartDraw.com](https://SmartDraw.com) - and also looking at [Lucid.co](https://lucid.co/) and will evaluate others. If we can leverage some of these amazing tools it will really help!

I will be sending a Pull Request that matches this flowchart as soon as I can.

**Comment by mattsayre** [comment]
@webaissance additional translation:

Your Results = Sus Resultados
 
Download Speed = Velocidad de Descarga
 
Upload Speed = Velocidad de Carga
 
Ping = Ping

**Comment by mattsayre** [comment]
@webaissance while we work to figure why the 'provider' lookup is not working, let's add IP Address to the csv export, so that it can be derived later in the overall workflow (in the separate analysis tool).

**Comment by mattsayre** [comment]
@webaissance in the CSV export, the significant digits for lat/long appear to be truncating or rounding. Let's work together to ensure that those get read out fully.

**Comment by webaissance** [comment]
Oh - thanks for catching that. I'll look into fixing it.
